bug(compiler/developers): lexical model compiler improperly distributed on alpha, beta publishing tiers.

**Describe the bug**
It is impossible to `npm install @keymanapp/lexical-model-compiler@beta` (or `@alpha`) due to improperly specified dependencies for those publishing tiers.

In a screenshot:

<img width="697" alt="Screen Shot 2021-01-19 at 10 10 30 AM" src="https://user-images.githubusercontent.com/25213402/104983191-de5c5f80-5a3e-11eb-9474-173094fbf162.png">

Note:

```
+ @keymanapp/models-types@14.0.224-beta
added 1 package from 3 contributors and audited 23 packages in 1.66s
found 0 vulnerabilities
```

But:

```
jahorton@Joshuas-MBP lexical-models % npm install @keymanapp/lexical-model-compiler@beta
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @keymanapp/models-types@^14.0.224.
```

That `npm install` failed because no `14.0.224` version was available - only `14.0.224-beta`.  Thus, when publishing to a tagged tier type, package.json should specify that same publishing tier for any in-repo dependencies.

I found this while attempting to update the https://github.com/keymanapp/lexical-models repo in order to address https://github.com/keymanapp/lexical-models/issues/108 and #4115, as the new casing feature requires a beta-version of the lexical-model compiler.

**To Reproduce**

1. `npm install @keymanapp/lexical-model-compiler@beta`
2. See error

**Expected behavior**
A healthy, error-free `npm install`.

--- a/resources/shellHelperFunctions.sh
+++ b/resources/shellHelperFunctions.sh
@@ -251,6 +251,10 @@ set_version ( ) {
 #   set_npm_version VERSION_WITH_TIER
 #
 set_npm_version () {
+  if [ $# == 0 ]; then
+    fail "set_npm_version requires a specified version."
+  fi
+  
   local version=$1
   # We use --no-git-tag-version because our CI system controls version numbering and
   # already tags releases. We also want to have the version of this match the

--- a/resources/shellHelperFunctions.sh
+++ b/resources/shellHelperFunctions.sh
@@ -257,7 +257,10 @@ set_npm_version () {
   # release of Keyman Developer -- these two versions should be in sync. Because this
   # is a large repo with multiple projects and build systems, it's better for us that
   # individual build systems don't take too much ownership of git tagging. :)
-  npm --no-git-tag-version --allow-same-version version "$version" || fail "Could not set package version to $version."
+  pushd . > /dev/null
+  cd "${KEYMAN_ROOT}"
+  npm run version "$version"  || fail "Could not set package versions to $version."
+  popd > /dev/null
 }
 
 # Initializes use of the npm `lerna` package within the repo.

